sdk/tables/azure-data-tables/src/test/java/com/azure/data/tables/autorestTest.java

```diff
@@ -0,0 +1,364 @@
+            authSettings.getAccount().getAccessKey());
+
+        final List<HttpPipelinePolicy> policies = Arrays.asList(
+            new AddDatePolicy(),
+            new AddHeadersPolicy(new HttpHeaders().put("Accept", OdataMetadataFormat.APPLICATION_JSON_ODATA_MINIMALMETADATA.toString())),
+            new TablesSharedKeyCredentialPolicy(sharedKeyCredential),
+            new HttpLoggingPolicy(new HttpLogOptions()
+                .setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
+        );
+        final HttpPipeline pipeline = new HttpPipelineBuilder()
+            .policies(policies.toArray(new HttpPipelinePolicy[0]))
+            .build();
+        AzureTableImpl azureTable = new AzureTableImplBuilder()
+            .pipeline(pipeline)
+            .version("2019-02-02")
+            .url("https://telboytrial.table.core.windows.net")
+            .buildClient();
+        return azureTable;
+    }
+
+
+    @BeforeEach
+    void beforeTests() {
+        createTableHelper(tableA);
+        createTableHelper(tableB);
+
+        propertiesB.put("PartitionKey", "Store");
+        propertiesB.put("RowKey", "Boston");
+        propertiesB.put("Employees", "200");
+        insertEntityHelper(tableA, propertiesB);
+
+        Map<String, Object> propertiesA = new HashMap<>();
+        propertiesA.put("PartitionKey", "Store");
+        propertiesA.put("RowKey", "Atlanta");
+        propertiesA.put("Employees", "50");
+        insertEntityHelper(tableA, propertiesA);
+
+
+    }
+
+    @AfterEach
+    void afterTests() throws InterruptedException {
+        deleteTableHelper(tableA);
+        deleteTableHelper(tableB);
+
+    }
+
+    void createTableHelper(String tableName) {
+        AzureTableImpl azureTable = auth();
+
+        TableProperties tableProperties = new TableProperties().setTableName(tableName);
+        String requestId = UUID.randomUUID().toString();
+
+        azureTable.getTables().createWithResponseAsync(tableProperties, requestId,
+            ResponseFormat.RETURN_CONTENT, null, Context.NONE).block();
+
+    }
+
+    void deleteTableHelper(String tableName) throws InterruptedException {
+
+        AzureTableImpl azureTable = auth();
+        String requestId = UUID.randomUUID().toString();
+
+
+        azureTable.getTables().deleteWithResponseAsync(tableName, requestId,
+            Context.NONE).block();
+        TimeUnit.SECONDS.sleep(2);
+
+    }
+
+    void insertEntityHelper(String tableName, Map<String, Object> properties) {
+        String requestId = UUID.randomUUID().toString();
+
+        azureTable.getTables().insertEntityWithResponseAsync(tableName, 500,
+            requestId, ResponseFormat.RETURN_CONTENT, properties, null, Context.NONE).log().block();
+    }
+
+    @Test
+    void createTable() throws InterruptedException {
+        TableProperties tableProperties = new TableProperties().setTableName(tableZ);
```

Comment:
These tests should follow arrange, act, and assert. All declarations should be done before your Act & Assert steps.
https://docs.telerik.com/devtools/justmock/basic-usage/arrange-act-assert#verify-interaction

An example:
https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusSenderClientTest.java#L165